sklearn/model_selection/_split.py

```diff
@@ -287,9 +287,17 @@ def __init__(self, n_splits, shuffle, random_state):
             raise TypeError("shuffle must be True or False;"
                             " got {0}".format(shuffle))
 
+        if not shuffle and random_state is not None:  # None is the default
+            raise ValueError(
+                'Setting a random_state will have no effect when shuffle is '
+                'False.'
+            )
+
         self.n_splits = n_splits
         self.shuffle = shuffle
-        self.random_state = random_state
+        self.random_state = random_state  # only here for repr to work
```

Comment:
What are the benefits of having the repr work? With this PR, we do not need to hold on to `self.random_state`.

Reply:
> What are the benefits of having the repr work?

CI and tests not breaking ^^

If we remove the `random_state` attribute, it is rendered as `random_state=None` (no matter the value) by the custom `repr`.

I feel like a reasonable fix would be to print the seed attr instead of the random_state, but didn't want to involve too many changes in the PR.